crates/bevy_hierarchy/src/hierarchy.rs

```diff
@@ -131,7 +208,38 @@ impl<'w> DespawnRecursiveExt for EntityWorldMut<'w> {
         .entered();
 
         self.world_scope(|world| {
-            despawn_children_recursive(world, entity);
+            despawn_children_recursive(world, entity, true);
+        });
+        self
+    }
+
+    /// Despawns the provided entity and its children.
+    /// This will emit warnings for any entity that does not exist.
+    fn try_despawn_recursive(self) {
+        let entity = self.id();
+
+        #[cfg(feature = "trace")]
+        let _span = bevy_utils::tracing::info_span!(
+            "try_despawn_recursive",
+            entity = bevy_utils::tracing::field::debug(entity)
```

Comment:
here, rather than creating 2 different spans for warn/no warn, we can just add a field to the span